File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const freqMap = nums.reduce((pre, curr) => {
            if (pre[curr] !== undefined) {
                pre[curr]++
            } else {
                pre[curr] = 1
            }
            return pre
        }, {})

        const freqs = Object.values(freqMap).sort((a, b) => b - a).filter((_, index) => index < k)

        const result = nums.reduce((pre, curr) => {
            if (!pre.includes(curr) && freqs.includes(freqMap[curr])) {
                pre.push(curr)
            }
            return pre
        }, [])

        return result
    }
}

File: Data Structures & Algorithms/jump-game/submission-2.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp = [False] * n
        dp[0] = True

        for i in range(n - 1):
            if dp[i] == False:
                return False
                
            k = nums[i]
            while k:
                if i + k > n - 1:
                    k -= 1
                    continue
                dp[i + k] = True
                k -= 1
        
        return dp[n - 1]